### PR TITLE
Refactor/migrate to typescript

### PR DESCRIPTION
**Pull request by uqmessias**
This PR solves the #10

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -25,22 +25,22 @@ import * as utils from './utils/transformers';
...
+  path: Path,
+  polygon: Polygon,
+  polyline: Polyline,
+  radialGradient: RadialGradient,
+  rect: Rect,
+  stop: Stop,
+  svg: Svg,
+  text: Text,
+  tspan: TSpan,
+  use: Use,
```
Comment (fjcaetano) [inline src/index.js]: 💯

File: index.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+import SvgRenderer from 'src';
+export default SvgRenderer;
```
Comment (fjcaetano) [inline index.js]: Do you still need an `index.js`?

File: src/utils/transformers/index.ts [hunk 3]
```diff
@@ -24,16 +33,15 @@ export const transformStyle = ({ name, value, fillProp }) => {
         [dashToCamelCase(property)]:
           fillProp && property === 'fill' ? fillProp : propertyValue,
       };
-    }, {});
+    }, {} as Record<string, any>);
```
Comment (fjcaetano) [inline src/utils/transformers/index.ts]: Could this be `unknown` instead of `any`?

File: src/utils/transformers/index.ts [hunk 4]
```diff
@@ -178,21 +192,21 @@ export const elementsMap = {
...
-    postProcessAttributes: (attributes, props, node) => {
+    postProcessAttributes: (attributes?: Attributes, props?: Props, node?: XmlNode) => {
       const { height, width } = props || {};
 
       if (!height && !width) {
         return attributes;
       }
 
-      const attrs = Object.assign({}, attributes);
+      const attrs = Object.assign({} as Partial<Props>, attributes);
```
Comment (fjcaetano) [inline src/utils/transformers/index.ts]: This is the same as `{ ... attributes }`

File: src/utils/transformers/index.ts [hunk 5]
```diff
@@ -178,21 +192,21 @@ export const elementsMap = {
...
-        attrs.height = height;
+      if (height !== undefined) {
+        attrs.height = height?.toString();
       }
 
-      if (width) {
-        attrs.width = width;
+      if (width !== undefined) {
+        attrs.width = width?.toString();
       }
```
Comment (fjcaetano) [inline src/utils/transformers/index.ts]: This is a bit redundant. You don't need the `ifs` AND the optional chaining:
```ts
attrs.height = height?.toString() ?? attrs.height;
attrs.width = width?.toString() ?? attrs.width;
```

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    },
+    "strict": true,
+    "target": "esnext"
+  },
+  "exclude": [
+    "node_modules",
+    "dist",
+    "babel.config.js",
+    "metro.config.js",
+    "jest.config.js",
```
Comment (fjcaetano) [inline tsconfig.json]: Do you really need to explicitly exclude JS files? This seems weird.

File: tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    "src/**/__tests__",
+    "src/**/__mocks__",
```
Comment (fjcaetano) [inline tsconfig.json]: Are you sure you want to ignore typings for tests and mocks?

File: .npmignore [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+coverage/
+example/
+src/
+babel.config.js
+global.d.ts
+jest.config.js
+tsconfig.json
+
+**/__tests__/
+**/__mocks__/
```
Comment (fjcaetano) [inline .npmignore]: A safer approach to using `.npmignore` is to configure a "files" attribute in your package. This will ensure only the files described there + README, package and license will be exported with your package. You can always test how the package will look like by running `npm pack`

File: package.json [hunk 9]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "react-native-svg-renderer",
   "version": "0.1.2",
-  "description": "Render an SVG Image from an URL",
+  "description": "Render a SVG Image from an URL",
```
Comment (fjcaetano) [inline package.json]: What about "Renders SVG Images from URLs"?

File: src/index.tsx [hunk 10]
```diff
@@ -216,7 +216,7 @@ class SvgRenderer extends React.PureComponent<Props, State> {
       );
     }
 
-    return null;
+    return <View />;
```
Comment (fjcaetano) [inline src/index.tsx]: Why did you add this change? This will actively add and render a component in the view stack, while returning `null` wouldn't render anything.

**Review by fjcaetano — APPROVED** [review]
Good stuff. Added minor comments

File: index.js (hunk 2 above)
Comment (uqmessias) [reply index.js]: I was using it to test, but it seems we don't need it anymore.

File: src/index.tsx (hunk 10 above)
Comment (uqmessias) [reply src/index.tsx]: Because when using the `SvgRenderer` along with animations (`Animated.createAnimatedComponent(SvgRenderer)`) the view is detached and causes a crash. This was the only way I could make it work properly. Do you have any other ideas of how to fix that without returning a `<View />`?

File: src/utils/transformers/index.ts (hunk 5 above)
Comment (uqmessias) [reply src/utils/transformers/index.ts]: Good catch!

File: tsconfig.json (hunk 6 above)
Comment (uqmessias) [reply tsconfig.json]: This was before removing `allowJs`, but I'll remove these excludes.

File: tsconfig.json (hunk 7 above)
Comment (uqmessias) [reply tsconfig.json]: This isn't ignoring the typing, it will only not compile them and generate `.js` files from them.

File: package.json (hunk 9 above)
Comment (uqmessias) [reply package.json]: Good idea!

Comment (uqmessias) [reply package.json]: What do you think about "Renders SVG Images from URLs and plain SVG string"?

File: .npmignore (hunk 8 above)
Comment (uqmessias) [reply .npmignore]: I tried that before, but for some weird reason, the result of ts compilation included the `src` files inside `dist`.
I updated it here and worked well.
